src/server/snapshot.cc

```diff
@@ -95,246 +94,242 @@ void SliceSnapshot::Join() {
     snapshot_fb_.join();
 }
 
+// The algorithm is to go over all the buckets and serialize those with
+// version < snapshot_version_. In order to serialize each physical bucket exactly once we update
+// bucket version to snapshot_version_ once it has been serialized.
+// We handle serialization at physical bucket granularity.
+// To further complicate things, Table::Traverse covers a logical bucket that may comprise of
+// several physical buckets in dash table. For example, items belonging to logical bucket 0
+// can reside in buckets 0,1 and stash buckets 56-59.
+// PrimeTable::Traverse guarantees an atomic traversal of a single logical bucket,
+// it also guarantees 100% coverage of all items that exists when the traversal started
+// and survived until it finished.
+
 // Serializes all the entries with version less than snapshot_version_.
-void SliceSnapshot::SerializeEntriesFb(const Cancellation* cll) {
-  this_fiber::properties<FiberProps>().set_name(
-      absl::StrCat("SliceSnapshot", ProactorBase::GetIndex()));
-  PrimeTable::Cursor cursor;
+void SliceSnapshot::IterateBucketsFb(const Cancellation* cll) {
+  {
+    auto fiber_name = absl::StrCat("SliceSnapshot-", ProactorBase::GetIndex());
+    this_fiber::properties<FiberProps>().set_name(std::move(fiber_name));
+  }
 
+  PrimeTable::Cursor cursor;
   for (DbIndex db_indx = 0; db_indx < db_array_.size(); ++db_indx) {
-    if (cll->IsCancelled())
-      return;
-
-    if (!db_array_[db_indx])
+    if (cll->IsCancelled() || !db_array_[db_indx])
```

Comment:
before if cll->IsCancelled() == true we returned from function. Why did you change to continue?

Reply:
I just changed it to be more readable. Its slightly less efficient now, but it should work in both cases. But I'm not sure if its worth the trade-off

Reply:
I am not sure this specific change is more readable
in this case it actually will raise a question when reading the flow why if it cancelled you continue and not return, and than will require someone to dig in the flow to answer his question and this is less good in my opinion

Reply:
fixed